text/0079-cdk-2.0.md

```diff
@@ -0,0 +1,384 @@
+annotated code, as well as any private function that is no longer used. In the event non-`@deprecated` code is found to
+depend on `@deprecated` code, the implementation in `master` will be fixed to no longer defer into the `@deprecated`
+feature, and that fix will be forward-ported into the `next` branch in the exact same way that other changes are.
+
+### Resetting v1 feature flags
+
+When developing new behaviors that are incompatible with previous ones, feature flags were introduced to allow customers
+control of when they migrate over to the _new behavior_. However, when a new major version release is issued, the _new
+behavior_ becomes the _only behavior_.
+
+In order to reduce the codebase divergence between `master` and `next` (in order to minimize the amount of manual work
+involved in forward-porting features on `next`), existing feature flags must be made version-aware, such that they are
+forcefully enabled when the current version is greater then `2.0.0-0`.
+
+The actual removal of code paths that support the _old behavior_ will be deferred to _after_ the new major version has
+reached _General Availability_.
+
+### Monolithic Packaging
+
+The AWS Construct Library will be released as a single artifact, instead of a collection of more than 100 libraries. The
+`monocdk-experiment` package has the tools necessary to re-package the existing _hypermodular_ libraries into a single
+packaging. Re-using that same process will allow keeping the difference between `master` and `next` as constrained as
+possible (reducing the need for manual intervention in forward-porting features).
+
+Consequently, the `monocdk-experiment` package will be renamed so that it is published and advertised as
+`@aws-cdk/stdlib` on the `master` branch. Availability of this library as part of CDK v1 is going to provide customers
+with a stepping stone to begin migrating their applications to CDK v2.
+
+Additionally, all modules that are re-packaged as part of `@aws-cdk/stdlib` will be marked as `"private": true` in the
+`next` branch, making these available _only_ via `@aws-cdk/stdlib` in CDK v2.
+
+## Documentation Updates
+
+Before formally announcing the availability of CDK v2 for early adopters, a documentation source must be availble to
+facilitate use of the new repository. The minimum bar should be availability of an API reference documentation for the
+v2 codebase that is usable.
+
+At the same time, the work to update the [AWS CDK User Guide] should be scoped and planned out. The updates to the User
+Guide must also include a migration guide to help developers move from CDK v1 over to v2. Special attention will be
+spent in ensuring the v2 release notes contain a clear and accurate summary of breaking changes and how users should
+update their applications.
+
+[aws cdk user guide]: https://docs.aws.amazon.com/cdk/latest/guide/home.html
+
+## Prerelease Publishing
```

Comment:
I think we should begin pre-release publishing as soon as we fork (or close to it) so it will be possible to test it in the wild. We don't have to announce it right away, but it's definitely something we should be able to point people to so they can give us early feedback. Trying to learn some lessons from our work on mono-cdk where we postponed this too long. For example, we will migrate delivlib and other internal projects to it.

There is no reason to wait for the documentation fix.